pkg/agent/agent.go

```diff
@@ -114,7 +119,19 @@ func (a *AgentOptions) RunAddOns(ctx context.Context) error {
 		return err
 	}
 
-	if err := helpers.EnsureCRDs(ctx, crdClient, manifests.AgentCRDFiles, requiredCRDFiles...); err != nil {
+	if err := helpers.EnsureCRDs(ctx, crdClient, manifests.AgentCRDFiles, agentRequiredCRDFiles...); err != nil {
+		return err
+	}
+
+	hostingKubeConfig, err := rest.InClusterConfig()
```

Comment:
nit: it's better to have a comment, in hosted mode, this kubeconfig is from management cluster, in default mode, this kubeconfig is from managed cluster

Reply:
for spokeKubeConfig, it always from managed cluster

Reply:
Done